pyincore/analyses/combinedwindwavesurgebuildingloss/combinedwindwavesurgebuildingloss.py

```diff
@@ -0,0 +1,209 @@
+
+                Returns:
+                    pd.DataFrame: An table of structural and content loss for each building
+
+                """
+        # Rename the columns so there are no overlapping names among the hazard damage states
+        flood_dmg.rename(columns={'LS_0': 'f_LS_0', 'LS_1': 'f_LS_1', 'LS_2': 'f_LS_2', 'DS_0': 'f_DS_0',
+                                  'DS_1': 'f_DS_1', 'DS_2': 'f_DS_2', 'DS_3': 'f_DS_3', 'haz_expose': 'f_haz_expose'},
+                         inplace=True)
+        
+        sw_dmg.rename(columns={'LS_0': 'sw_LS_0', 'LS_1': 'sw_LS_1', 'LS_2': 'sw_LS_2', 'DS_0': 'sw_DS_0',
+                               'DS_1': 'sw_DS_1', 'DS_2': 'sw_DS_2', 'DS_3': 'sw_DS_3', 'haz_expose': 
+                                   'sw_haz_expose'}, inplace=True)
+
+        wind_dmg.rename(columns={'LS_0': 'w_LS_0', 'LS_1': 'w_LS_1', 'LS_2': 'w_LS_2', 'DS_0': 'w_DS_0',
+                                 'DS_1': 'w_DS_1', 'DS_2': 'w_DS_2', 'DS_3': 'w_DS_3', 'haz_expose': 'w_haz_expose'},
+                        inplace=True)
+
+        # Combine all three sets of damages into a single data frame
+        combined_df = pd.merge(pd.merge(wind_dmg, sw_dmg, on='guid'), flood_dmg, on='guid')
+
+        # Add the flood archetype to the combined damage since this is used to find loss multipliers
+        new_combined_df = pd.merge(combined_df, buildings[["guid", "arch_flood"]], on="guid")
+
+        # Create a result data frame for the loss calculations
+        loss_df = new_combined_df[['guid']].copy()
+
+        # Compute content and structural loss
+        loss_df['cont_loss'] = new_combined_df.apply(lambda row: np.dot(row[['f_DS_0', 'f_DS_1', 'f_DS_2']],
+                                                                        content_cost.iloc[int(row['arch_flood']-1),
+                                                                        1:4]), axis=1)
+
+        loss_df['roof_loss'] = new_combined_df.apply(lambda row: np.dot(row[['w_DS_0', 'w_DS_1', 'w_DS_2', 'w_DS_3']],
+                                                                        np.array([0.15, 0.5, 0.75, 1])) *
+                                                     structure_cost.loc[int(row['arch_flood']-1), "Roofing"], axis=1)
+
+        loss_df['ff_loss'] = new_combined_df.apply(lambda row: np.dot(row[['sw_DS_0', 'sw_DS_1', 'sw_DS_2',
+                                                                           'sw_DS_3']], np.array([0.1, 0.5, 0.75, 1])) *
+                                                   structure_cost.loc[int(row['arch_flood'] - 1),
+                                                   "Flooring and Foundation"], axis=1)
+
+        # Computes frame loss from the dominant hazard between wind and surge-wave intensities
+        def compute_frame_loss(row):
+            if row['w_DS_3'] >= row['sw_DS_3']:
+                return np.dot(row[['w_DS_0', 'w_DS_1', 'w_DS_2', 'w_DS_3']],np.array([0.25, 0.5, 0.75, 1])) * \
```

Comment:
Are those number fixed? 0.25, 0.5 .. i guess we are dividing 1/4 cuz we have 4 ds..

Reply:
Yes, they were provided by Jiate. Maybe next release we can see if those should be inputs or if they are fixed.